moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js

```diff
@@ -0,0 +1,126 @@
+    // todo: actually store the traffic signal ID somewhere
+    // probably as new component attributes "asset_id" and "asset_type"
+    const feature = {
+      type: "Feature",
+      properties: {
+        ...signal.properties,
+        renderType: "Point",
+        PROJECT_EXTENT_ID: featureUUID,
+        sourceLayer: "drawnByUser",
+      },
+      geometry: signal.geometry,
+      id: featureUUID,
+    };
+    featureCollection.features.push(feature);
+  }
+  return featureCollection;
+};
+
+const SignalAutocomplete = ({
+  signal,
+  setSignal,
+  projectDetails,
+  setProjectDetails,
+  setFeatureCollection,
+  nameError,
+}) => {
+  const [data, setData] = React.useState(null);
+  const [error, setError] = React.useState(null);
+
+  const handleFieldChange = signal => {
+    const projectName = signal?.properties?.location_name || "";
+    const updatedProjectDetails = {
+      ...projectDetails,
+      ...{ project_name: projectName },
+    };
+    setProjectDetails(updatedProjectDetails);
+    setSignal(signal);
+    const featureCollection = signalToFeatureCollection(signal);
+    setFeatureCollection(featureCollection);
+  };
+  const loading = !data && !error;
+
+  React.useEffect(() => {
+    const url =
+      "https://data.austintexas.gov/resource/p53x-x73x.geojson?$select=signal_id,location_name,location,signal_type&$order=signal_id asc&$limit=9999";
+    fetch(url)
+      .then(response => response.json())
+      .then(
+        result => {
+          if (result.error) {
+            // on query error, socrata returns status 200 with {"error": true, "message": <message>} in body
+            setError(result.message.toString());
+          } else {
+            // remember: socrata geojson request returns a featureCollection
+            setData(result.features);
+          }
+        },
+        error => {
+          setError(error.toString());
+        }
+      );
+  }, []);
+
+  return (
+    <Autocomplete
+      id="signal-id"
+      // func to match the value from props to the selected option we should display
+      getOptionSelected={(option, value) => {
+        // todo: i had to use optional chaning here, but i'm not sure why. the `value` test was
+        // seemingly calling the first condition when value was ""
+        return value
+          ? option.properties?.signal_id === value.properties?.signal_id
+          : option === "";
+      }}
+      getOptionLabel={option =>
+        option
+          ? `${option.properties.signal_id}: ${option.properties.location_name}`
+          : ""
+      }
+      onChange={(e, signal) => {
+        handleFieldChange(signal);
+      }}
+      loading={loading}
+      options={data || []}
+      renderInput={params => (
+        <TextField
+          {...params}
+          error={nameError}
+          helperText="Required"
+          InputProps={{
+            ...params.InputProps,
+            endAdornment: (
+              <>
+                {loading ? (
```

Comment:
it's hard to verify the spinner renders when testing because the Socrata request (probably cachced) is faaast :) i've seen it, I swear.